ruby-challenges.rb

```diff
@@ -19,11 +27,34 @@
 nums_array2 = [25, 17, 47, 11]
 # Expected output: 100
 
+def add_number array
+    array.sum
+end 
+
+p add_number nums_array1
+p add_number nums_array2
+
 
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
-# Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
+class Bike 
+    attr_reader :current_speed
+    def initialize(model, wheels=2, current_speed=0)
+        @model = model
+        @wheels = wheels
+        @current_speed = current_speed
+    end
+
+    def bike_info
+        p "The #{@model} bike has #{@wheels} wheels and is going #{@current_speed} mph"
+    end
+end
+
+tricycle = Bike.new('Tricycle')
+tricycle.bike_info
+
+Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
```

Comment:
It is a best practice to always use the `p` on the method invocation rather than on the inner workings of your method.